pio/ws2812/ws2812.c

```diff
@@ -54,44 +54,44 @@ static inline uint32_t urgbw_u32(uint8_t r, uint8_t g, uint8_t b, uint8_t w) {
             (uint32_t) (b);
 }
 
-void pattern_snakes(uint len, uint t) {
+void pattern_snakes(PIO pio, uint sm, uint len, uint t) {
     for (uint i = 0; i < len; ++i) {
         uint x = (i + (t >> 1)) % 64;
         if (x < 10)
-            put_pixel(urgb_u32(0xff, 0, 0));
+            put_pixel(pio, sm, urgb_u32(0xff, 0, 0));
         else if (x >= 15 && x < 25)
-            put_pixel(urgb_u32(0, 0xff, 0));
+            put_pixel(pio, sm, urgb_u32(0, 0xff, 0));
         else if (x >= 30 && x < 40)
-            put_pixel(urgb_u32(0, 0, 0xff));
+            put_pixel(pio, sm, urgb_u32(0, 0, 0xff));
         else
-            put_pixel(0);
+            put_pixel(pio, sm, 0);
     }
 }
 
-void pattern_random(uint len, uint t) {
+void pattern_random(PIO pio, uint sm, uint len, uint t) {
     if (t % 8)
         return;
     for (uint i = 0; i < len; ++i)
-        put_pixel(rand());
+        put_pixel(pio, sm, rand());
 }
 
-void pattern_sparkle(uint len, uint t) {
+void pattern_sparkle(PIO pio, uint sm, uint len, uint t) {
     if (t % 8)
         return;
     for (uint i = 0; i < len; ++i)
-        put_pixel(rand() % 16 ? 0 : 0xffffffff);
+        put_pixel(pio, sm, rand() % 16 ? 0 : 0xffffffff);
 }
 
-void pattern_greys(uint len, uint t) {
+void pattern_greys(PIO pio, uint sm, uint len, uint t) {
     uint max = 100; // let's not draw too much current!
     t %= max;
     for (uint i = 0; i < len; ++i) {
-        put_pixel(t * 0x10101);
+        put_pixel(pio, sm, t * 0x10101);
         if (++t >= max) t = 0;
     }
 }
 
-typedef void (*pattern)(uint len, uint t);
+typedef void (*pattern)(PIO pio, uint sm, uint len, uint t);
```

Comment:
I'm 99% sure that it doesn't matter, but just making an observation here that this now means that `ws2812_parallel.c` now has a different typedef
```c
typedef void (*pattern)(uint len, uint t);
```
to the one in `ws2812.c`.

Reply:
None of the functions reference the pio or sm in ws2812_parallel.c because they're using dma